app/code/Magento/InventorySales/Model/IsProductSalableCondition/IsSalableLegacyStockItemIsInStock.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Model\IsProductSalableCondition;
+
+use Magento\InventorySalesApi\Api\IsProductSalableInterface;
+use Magento\CatalogInventory\Api\StockConfigurationInterface;
+use Magento\CatalogInventory\Model\Spi\StockRegistryProviderInterface;
+use Magento\InventoryCatalogApi\Model\GetProductIdsBySkusInterface;
+use Magento\InventoryConfiguration\Model\IsSourceItemManagementAllowedForSku;
+
+/**
+ * @inheritdoc
+ */
+class IsSalableLegacyStockItemIsInStock implements IsProductSalableInterface
```

Comment:
can you re-name the condition?
Why does it contain word Legacy ?

We will have the same case even with brand new Inventory mechanism, as here we add handling for composite product types